src/packages/solid/components/Icon/Icon.tsx

```diff
@@ -46,15 +46,20 @@ export interface IconProps extends IntrinsicNodeProps {
 }
 
 const Icon: Component<IconProps> = props => {
+  let IconRef;
+  console.log(IconRef);
   return (
     <View
       {...props}
       tone={props.tone ?? styles.tone}
       style={[
+        ...props.style,
         props.style?.Container,
-        props.style?.Container[props.tone || styles.tone],
+        props.style?.Container?.[props.tone || styles.tone],
```

Comment:
with out it there were errors with it thinking its a ternary

Reply:
its still not valid JS (something must be off. But I did want to argue that we shouldn't be doing props style with additional tone styling. It's overkill

Reply:
The styles are getting passed into the icon as an array rather than merging them becasue the merge props seems to only work on a view, so within the Icon I was trying to handle the way the props had to merge